core/src/main/java/se/tre/freki/core/MetaClient.java

```diff
@@ -198,13 +199,18 @@ public void deleteAnnotation(final Annotation note) {
    */
   public ListenableFuture<Boolean> update(final LabelMeta meta) {
     return transform(getLabelMeta(meta.type(), meta.identifier()),
-        new AsyncFunction<LabelMeta, Boolean>() {
+        new AsyncFunction<Optional<LabelMeta>, Boolean>() {
           @Override
-          public ListenableFuture<Boolean> apply(final LabelMeta storedMeta) throws Exception {
-            if (!storedMeta.equals(meta)) {
-              return store.updateMeta(meta);
+          public ListenableFuture<Boolean> apply(final Optional<LabelMeta> storedMeta)
+              throws Exception {
+            if (!storedMeta.isPresent()) {
+              LOG.debug("{} was not found.", meta);
```

Comment:
Can you raise the log level here to `warn`? If the user tries to update a meta that is missing it might be nice to give a bigger splash about it being missing.